internal/messagejournal/messagejournal.go

```diff
@@ -0,0 +1,216 @@
+	if err != nil {
+		return nil, fmt.Errorf("database object not created: %w", err)
+	}
+	messageJournal := MessageJournal{database: db}
+
+	err = db.Ping()
+	if err != nil {
+		return nil, fmt.Errorf("message journal database not connected: %w", err)
+	}
+
+	if _, err = db.Exec(messageJournalTemporaryDropString); err != nil {
+		return nil, fmt.Errorf("runtime messsage journal table not deleted: %w", err)
+	}
+	if _, err = db.Exec(messageJournalTemporaryCreateString); err != nil {
+		return nil, fmt.Errorf("runtime messsage journal table not created: %w", err)
+	}
+	if _, err = db.Exec(messageJournalPersistentCreateString); err != nil {
+		return nil, fmt.Errorf("persistent messsage journal table not created: %w", err)
+	}
+
+	return &messageJournal, nil
+}
+
+// Adds a new message journal entry to both the temporary runtime table
+// which gets cleared at the start of every session and the persistent table
+// which maintains message entries across multiple sessions.
+func (j *MessageJournal) AddEntry(entry yggdrasil.WorkerMessage) (*yggdrasil.WorkerMessage, error) {
+	// If the message journal is nil, it means the message journal
+	// is not enabled in the config.toml file or provided as a user argument.
+	if j == nil {
+		return nil, fmt.Errorf("message journal not enabled")
+	}
+
+	runtimeAction, err := j.database.Prepare("INSERT INTO runtime(message_id, sent, worker_name, response_to, worker_event, worker_message) values (?,?,?,?,?,?)")
+	if err != nil {
+		return nil, err
+	}
+	persistentAction, err := j.database.Prepare("INSERT INTO persistent(message_id, sent, worker_name, response_to, worker_event, worker_message) values (?,?,?,?,?,?)")
+	if err != nil {
+		return nil, err
+	}
+
+	runtimeResult, err := runtimeAction.Exec(entry.MessageID, entry.Sent, entry.WorkerName, entry.ResponseTo, entry.WorkerEvent.EventName, entry.WorkerEvent.EventMessage)
+	if err != nil {
+		return nil, err
+	}
+	persistentResult, err := persistentAction.Exec(entry.MessageID, entry.Sent, entry.WorkerName, entry.ResponseTo, entry.WorkerEvent.EventName, entry.WorkerEvent.EventMessage)
+	if err != nil {
+		return nil, err
+	}
+
+	runtimeEntryID, err := runtimeResult.LastInsertId()
+	if err != nil {
+		return nil, fmt.Errorf("%v-%v", runtimeEntryID, err)
+	}
+	presistentEntryID, err := persistentResult.LastInsertId()
+	if err != nil {
+		return nil, fmt.Errorf("%v-%v", presistentEntryID, err)
+	}
+
+	return &entry, nil
+}
+
+func (j *MessageJournal) GetEntries(filter MessageJournalFilter) ([]map[string]string, error) {
+	// If the message journal is nil, it means the message journal
+	// is not enabled in the config.toml file or provided as a user argument.
+	if j == nil {
+		return nil, nil
+	}
+
+	entries := []map[string]string{}
+
+	// Build SQL query to retrieve journal entries.
+	// FIXME: It works but I hate it... is there a better
+	// way to do this without an external library?
+	args := []interface{}{}
+	queryString := "SELECT * FROM "
```

Comment:
Is there a way to create dynamic sql query strings in Go without bringing in an ORM or other external library? It looks messy constructing it with a series of conditional branches...

Reply:
It is good idea to use here something like this.

I'm sorry. I don't have such experience. I think that something like this could be useful, because this function is too long and it could help to refactor this code.